v2/apiserver/internal/api/mongodb/projects_store.go

```diff
@@ -205,16 +205,16 @@ function() {
 			{Key: "id", Value: 1},
 		},
 	)
-	cur, err := p.collection.Find(
-		ctx,
-		bson.M{
-			"spec.eventSubscriptions": bson.M{
-				"$elemMatch": preliminaryMatchCriteria,
-			},
-			"$where": where,
+	query := bson.M{
+		"spec.eventSubscriptions": bson.M{
+			"$elemMatch": preliminaryMatchCriteria,
 		},
-		findOptions,
-	)
+		"$where": where,
+	}
+	if event.ProjectID != "" {
+		query["id"] = event.ProjectID
+	}
+	cur, err := p.collection.Find(ctx, query, findOptions)
```

Comment:
Proactively addressing the reason for this change.

`ProjectList` (which is gone in favor of `meta.List[Project]`) used to have a `Contains()` function and if an event identified a specific project, what we used to do was find _all_ the subscribed projects and then check the service would call `projects.Contains(event.projectID)` to see if it was among them.

`meta.List` has no `Contains()` functions, but it hardly matters because what we were doing in the first place was really inefficient.

This change right here just adds the project ID to the query criteria if the event had specified one.